konfig

```diff
@@ -84,28 +85,46 @@ merge() {
 }
 
 import_ctx() {
+    declare -a tmpcfgs=()
     local tmpcfg
+    local tmpinputcfg=""
     local out=""
     local arg=""
-    if [[ "$1" == '-p' || "$1" == '--preserve-structure' ]]; then
-      arg="$1"
-      shift
-    fi
-    if [[ "$1" = '--save' || "$1" = '-s' ]]; then
-      out="${XDG_CACHE_HOME:-$HOME/.kube}/config"
-      shift
-    elif [[ "$1" =~ ^-(.*) ]]; then
-      error "unrecognized flag \"$1\""
-    fi
+    for OPT in "$@"; do
+      case $OPT in
+        -p | --preserve-structure)
+          arg="$1"
+          shift
+          ;;
+        -s | --save)
+          out="${XDG_CACHE_HOME:-$HOME/.kube}/config"
+          shift
+          ;;
+        -i | --stdin)
+          if [[ -p /dev/stdin ]]; then
+            tmpinputcfg=$(mktemp konfig_input_XXXXXX)
+            TMPFILES+=( "$tmpinputcfg" )
+            cat - > "$tmpinputcfg"
+            shift
+          fi
+          ;;
+        -*)
+          error "unrecognized flag \"$1\""
+          ;;
+      esac
+    done
     tmpcfg=$(mktemp konfig_XXXXXX)
     TMPFILES+=( "$tmpcfg" )
     $KUBECTL config view --raw > "$tmpcfg"
 
+    tmpcfgs+=( "$tmpcfg")
+    tmpcfgs+=( "$tmpinputcfg" )
```

Comment:
I'm wondering if this really works as intended. If we add an empty string to the array, doesn't that mean that there will be a trailing colon again?
Can you check this? (I haven't checked this yet)

If this is still a problem, the solution should be to append to the array in the options loop above.

Reply:
Thank you for pointing out this. You're right, the issue was exposed.
But to add `$tmpcfg` into the array is required for the current context of `$HOME/.kube` to take precedence.
So I use `if` not to face the issue.